File: code_samples/r/database_query_builder.r
# Database: Query Builder
# AI/ML Training Sample

QueryBuilder <- setRefClass(
    "QueryBuilder",
    fields = list(data = "character"),
    methods = list(
        initialize = function() {
            data <<- ""
        },
        process = function(input) {
            data <<- input
        },
        getData = function() {
            return(data)
        },
        validate = function() {
            return(nchar(data) > 0)
        }
    )
)

# Example usage
instance <- QueryBuilder$new()
instance$process("example")
cat("Data:", instance$getData(), "\n")
cat("Valid:", instance$validate(), "\n")

File: code_samples/scala/database_query_builder.scala
/**
 * Database: Query Builder
 * AI/ML Training Sample
 */

class QueryBuilder {
  private var data: String = ""
  
  def process(input: String): Unit = {
    data = input
  }
  
  def getData: String = data
  
  def validate: Boolean = data.nonEmpty
}

object QueryBuilderExample {
  def main(args: Array[String]): Unit = {
    val instance = new QueryBuilder()
    instance.process("example")
    println(s"Data: ${instance.getData}")
    println(s"Valid: ${instance.validate}")
  }
}
